### PR TITLE
Only handle addresses of the own ip family

### LINKED ISSUE #70113: kube-proxy IPVS spurious logs removing IPv6 address
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug
/sig network
/area ipvs
/area ipv6

**What happened**:

`kube-proxy` with IPVS mode creates a `kube-ipvs0` interface for binding service IPs.

```
ip addr show kube-ipvs0
5: kube-ipvs0: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default 
    link/ether 1a:ab:99:c0:a2:0d brd ff:ff:ff:ff:ff:ff
    inet 10.3.0.10/32 brd 10.3.0.10 scope global kube-ipvs0
       valid_lft forever preferred_lft forever
    inet 10.3.0.1/32 brd 10.3.0.1 scope global kube-ipvs0
       valid_lft forever preferred_lft forever
    inet6 fe80::18ab:99ff:fec0:a20d/64 scope link 
       valid_lft forever preferred_lft forever
```

kube-proxy periodically [tries to unbind](https://github.com/kubernetes/kubernetes/blob/v1.12.1/pkg/proxy/ipvs/proxier.go#L1583) addresses it doesn't know about from the interface. It tries to unbind the IPv6 address over and over.

```
Failed to unbind service addr fe80::18ab:99ff:fec0:a20d from dummy interface kube-ipvs0: error unbind address: fe80::18ab:99ff:fec0:a20d from interface: kube-ipvs0, err: cannot assign requested address
```

The error is a little vague, but see below. The IP parsing or address deletion code [here](https://github.com/kubernetes/kubernetes/blob/v1.12.1/pkg/proxy/ipvs/netlink_linux.go#L61) may be a suspect.

**What you expected to happen**:

Either ignore the IPv6 address on the interface (its not hurting anything) or remove it correctly. 

**How to reproduce it (as minimally and precisely as possible)**:

Its perfectly natural for a new interface to have an inet6 address by default.

```
sudo ip link add name eth99 type dummy
```

If for some reason your `kube-ipvs0` interface doesn't have a similar address, you can add one to repro.

```
sudo ip addr add fe80::4482:f7ff:fe36:3e6d/64 dev kube-ipvs0
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.12.1
- Cloud provider or hardware configuration: AWS, GCP, DigitalOcean, bare-metal, Azure
- OS (e.g. from /etc/os-release): Container Linux 1855.4.0
- Kernel (e.g. `uname -a`): 4.14.67-coreos
- Install tools: Typhoon

### PR DESCRIPTION
**Pull request by uablrek**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

This is an updated re-post of https://github.com/kubernetes/kubernetes/pull/71831 which kind of broke down after diverging with master.

**Which issue(s) this PR fixes**:

Fixes #70113

**Special notes for your reviewer**:

Please see; https://github.com/kubernetes/kubernetes/pull/71831

**Does this PR introduce a user-facing change?**:

NONE

```release-note
```

### CODE REVIEW

**Comment by m1093782566** [comment]
@uablrek 

What can we do for tests?

/lgtm

**Comment by uablrek** [comment]
@m1093782566 

I did repeat the ipv4-test for address cleanup but set the nodeIP in the proxier to `net.ParseIP("::1")` to fake ipv6-only.
With the latest updates it becomes a bit harder. I will try to make an ipv6-version of `TestCleanLegacyService` but it contains more than just address clean-up so I am unsure how much work that is needed.

**Comment by uablrek** [comment]
@m1093782566 

I am updating `proxier_test.go` it is not very complicated. I add some ipv6 addresses in `TestCleanLegacyService` and create a new `TestCleanLegacyService6` test for ipv6-only. I will probably commit in about an hour.

I do not understand where the *pull-kubernetes-integration* test fails. I see several connection fails but more than 2 as indicated as # of failed tests. Can you please tell me what to look for?

File: pkg/proxy/ipvs/proxier_test.go [hunk 1]
```diff
@@ -2890,15 +2891,125 @@ func TestCleanLegacyService(t *testing.T) {
...
+	for _, a := range remainingAddrs {
+		if net.ParseIP(a).To4() == nil {
+			continue
+		}
+		remainingAddrsMap[a] = true
+	}
+	if !reflect.DeepEqual(activeBindAddrs, remainingAddrsMap) {
+		t.Errorf("Expected remainingAddrsMap %v, got %v", activeBindAddrs, remainingAddrsMap)
+	}
+
```
Comment (m1093782566) [inline pkg/proxy/ipvs/proxier_test.go]: nit: unneeded newline?

**Comment by maorfr** [comment]
@m1093782566 hey!
Do you know which release will contain this fix?

Thanks a lot for your great work guys!

**Comment by uablrek** [comment]
@mvladev I think it is easier to cherry-pick the original PR to 1.12 ans 1.13; https://github.com/kubernetes/kubernetes/pull/71831

**Pull request merged**